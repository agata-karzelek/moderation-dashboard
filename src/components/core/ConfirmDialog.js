import React from "react";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import { useStyles } from "../../style/confirmDialog";
import { Typography } from "@material-ui/core";

const ConfirmDialog = (props) => {
  const { title, children, open, setOpen, onConfirm } = props;
  const classes = useStyles();

  return (
    <Dialog
      id="dialog"
      open={open}
      onClose={() => setOpen(false)}
      aria-labelledby="confirm-dialog"
      fullWidth
      maxWidth="xs"
      className={classes.confirmDialog}
    >
      <DialogTitle id="confirm-dialog">{title}</DialogTitle>
      <DialogContent id="content">{children}</DialogContent>
      <DialogActions>
        <Button
          id="confirm"
          onClick={() => {
            setOpen(false);
            onConfirm();
          }}
          className={classes.confirmButton}
        >
          <Typography>{props.actionMessage}</Typography>
        </Button>

        <Button
          id="cancel"
          variant="contained"
          onClick={() => setOpen(false)}
          className={classes.cancelButton}
        >
          Cancel
        </Button>
      </DialogActions>
    </Dialog>
  );
};
export default ConfirmDialog;
