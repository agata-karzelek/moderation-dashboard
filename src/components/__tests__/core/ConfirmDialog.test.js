import React from "react";
import { shallow } from "enzyme";
import ConfirmDialog from "../../core/ConfirmDialog";

const setOpenFn = jest.fn();
const setConfirmFn = jest.fn();

describe("Test Cases for Confirm Dialog", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <ConfirmDialog
        title="Are you sure?"
        children="You want to delete this user?"
        open={true}
        setOpen={setOpenFn}
        onConfirm={setConfirmFn}
      />
    );
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("initial Values", () => {
    expect(wrapper.find("#confirm-dialog").text()).toBe("Are you sure?");
    expect(wrapper.find("#content").text()).toBe("You want to delete this user?");
  });

  test("check functions", () => {
    expect(setOpenFn).toHaveBeenCalledTimes(0);
    expect(setConfirmFn).toHaveBeenCalledTimes(0);

    wrapper.find("#cancel").simulate("click");
    expect(setOpenFn).toHaveBeenCalledTimes(1);

    wrapper.find("#confirm").simulate("click");
    expect(setOpenFn).toHaveBeenCalledTimes(2);
    expect(setConfirmFn).toHaveBeenCalledTimes(1);

    wrapper.find("#dialog").getElement().props.onClose();
    expect(setOpenFn).toHaveBeenCalledTimes(3);
    expect(setConfirmFn).toHaveBeenCalledTimes(1);
  });
});
