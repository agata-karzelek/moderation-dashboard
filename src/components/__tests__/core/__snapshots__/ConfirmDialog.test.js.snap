// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Cases for Confirm Dialog Snapshot 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-labelledby="confirm-dialog"
  className="makeStyles-confirmDialog-3"
  fullWidth={true}
  id="dialog"
  maxWidth="xs"
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))
    id="confirm-dialog"
  >
    Are you sure?
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))
    id="content"
  >
    You want to delete this user?
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-confirmButton-1"
      id="confirm"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(Typography)) />
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-cancelButton-2"
      id="cancel"
      onClick={[Function]}
      variant="contained"
    >
      Cancel
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
